sweep/aggregator.go

```diff
@@ -103,27 +98,7 @@ func (c *inputCluster) createInputSets(
 		// If there are no positive yield inputs, we can stop here.
 		inputCount := len(txInputs.inputs)
 		if inputCount == 0 {
-			return sets, nil
-		}
-
-		// Check the current output value and add wallet utxos if
-		// needed to push the output value to the lower limit.
-		if err := txInputs.tryAddWalletInputsIfNeeded(); err != nil {
-			return nil, err
-		}
-
-		// If the output value of this block of inputs does not reach
-		// the dust limit, stop sweeping. Because of the sorting,
-		// continuing with the remaining inputs will only lead to sets
-		// with an even lower output value.
-		if !txInputs.enoughInput() {
```

Comment:
previously we'd stop adding to `sets` if `enoughInput` returned false. now we'll continue adding to `sets` and only _later_ call `enoughInput` in `NeedWalletInput` and potentially add wallet inputs. this seems like a change in behavior?

Reply:
This changes the behavior when the two conditions are met,
1. the number of inputs is equal to or greater than 100;
2. the 100th input and beyond are positive-yielding.
3. the first 99-input set needs a wallet input.

Suppose we have 100 inputs, all positive-yielding, previously we'd stop at the 99th input, and add one more wallet input. Then a second input set, which has one input, and adds one more wallet input.
Now we will add all 100 inputs into one set, and won't be able to add the one more wallet input due to the size limit.

Need to double check if the assumptions make sense, as if they are all positive yielding, is a wallet input even needed?

Reply:
This makes sense to me. If *all* inputs have a positive yield then it doesn't need a wallet input. As long as the feerate is selected properly, passing this check should be enough. Also we want to max out the number of inputs we batch to maximally amortize the cost of the transaction header's overhead.